src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -113,19 +130,29 @@ public VerificationResult verifyPhoneWithCode(Request req, Response res) {
         if (code == null) {
             logMessageAndHalt(req, 400, "Missing code from verify request.");
         }
-        if (otpUser.phoneNumber == null) {
+        if (otpUser.pendingPhoneNumber == null) {
             logMessageAndHalt(req, 404, "User must have valid phone number for SMS verification.");
         }
         // Check verification code with SMS service.
-        VerificationCheck check = NotificationUtils.checkSmsVerificationCode(otpUser.phoneNumber, code);
+        VerificationCheck check = NotificationUtils.checkSmsVerificationCode(otpUser.pendingPhoneNumber, code);
         if (check == null) {
             logMessageAndHalt(
                 req,
                 HttpStatus.INTERNAL_SERVER_ERROR_500,
                 "Unknown error encountered while checking SMS verification code"
             );
         }
-        // If the check is successful, status will be "approved"
-        return new VerificationResult(check);
+
+        // If the check is successful, status will be "approved".
+        VerificationResult verificationResult = new VerificationResult(check);
+        if (verificationResult.status.equals("approved")) {
+            // If the check is successful, update the OtpUser's phoneNumber and erase pendingPhoneNumber.
+            // TODO: Figure out adding a test for this update to the OtpUser record.
+            otpUser.phoneNumber = otpUser.pendingPhoneNumber;
+            otpUser.pendingPhoneNumber = null;
```

Comment:
We did not agree to implement this change from storing a single phone number to pending + phone number.

Reply:
This is to not overwrite an existing confirmed number.